packages/ember-metal/lib/events.js

```diff
@@ -126,11 +126,9 @@ export function removeListener(obj, eventName, target, method) {
     target = null;
   }
 
-  metaFor(obj).removeFromListeners(eventName, target, method, (...args) => {
-    if ('function' === typeof obj.didRemoveListener) {
-      obj.didRemoveListener(...args);
-    }
-  });
+  let func = ('function' === typeof obj.didRemoveListener) ?
+    obj.didRemoveListener.bind(obj) : ()=> {};
```

Comment:
* avoids creating wrapping function
* does not do `if check` each cycle
* does not convert arguments to array
* does not do work when `obj.didRemoveListener` missing